Assets/Scripts/Health/Lives.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DN.UI
+{
+	/// <summary>
+	/// Takes care of the lives the player has to be added to Canvas
+	/// </summary>
+	public class Lives : MonoBehaviour
+	{
+
+		public int currentLives
+		{
+			get { return hearts.Count; }
+		}
+
+		[SerializeField] private int lives = 3;
+		[SerializeField] private float size = 15;
+		private Sprite heart;
+		private List<GameObject> hearts;
+		private RectTransform canvas;
+
+		private void Start()
+		{
+			canvas = gameObject.GetComponent<RectTransform>();
+			hearts = new List<GameObject>();
+			heart = Resources.Load<Sprite>("Sprites/heart");
+
+			for (int i = 0; i < lives; i++)
+			{
+				GameObject g = new GameObject($"heart {i}");
+				g.AddComponent<Image>().sprite = heart;
+				RectTransform rTransform = g.GetComponent<RectTransform>();
+				g.transform.parent = transform;
```

Comment:
noem een object geen g, geef het een leesbare naam zodat het makkelijker is om te begrijpen wat voor object het is.